src/plugins/formulas/__tests__/formulas.spec.js

```diff
@@ -2097,4 +2097,155 @@ describe('Formulas general', () => {
       ['A5']
     ]);
   });
+
+  describe('cooperation with validation', () => {
+    it('should validate result of formula properly (opening and closing an editor)', async() => {
+      const hot = handsontable({
+        data: [
+          ['=B1+5', 2, '=D1', 'text']
+        ],
+        formulas: {
+          engine: HyperFormula
+        },
+        type: 'numeric'
+      });
+
+      selectCell(0, 0);
+      // Opening an editor
+      keyDown('enter');
+      // Closing the editor and saving changes.
+      keyDown('enter');
+
+      await sleep(0); // Validator is asynchronous.
```

Comment:
Also, I tried removing `onBeforeValidate` and I changed this to `sleep(100)` - all of the cases seem to pass fine.

Perhaps `onBeforeValidate` won't be needed at all, removing the need for the workaround and simplifying the logic a bit?

What am I missing?

Reply:
Huh, interesting! Yes, I've checked that an it works... Thus, I will simplify.